src/proxy-ui-api/frontend/src/store/modules/addClient.ts

```diff
@@ -134,6 +134,24 @@ export const mutations: MutationTree<AddClientState> = {
   },
 };
 
+// Filters out clients that have local relatives.
+// If the member owning the client or another subsystem
+// of the same member is already present locally,
+// the client is excluded.
+const excludeClientsWithLocalRelatives = (clients: Client[], localClients: Client[]): Client[] => {
+  return clients.filter((client: Client) => {
+    let showClient = true;
+    localClients.forEach((localClient: Client) => {
+      if (localClient.member_class === client.member_class &&
+          localClient.member_code === client.member_code &&
+          localClient.instance_id === client.instance_id) {
+        showClient = false;
+      }
+    });
+    return showClient;
+  });
```

Comment:
Although this code works correctly I think it could be optimized and made in more currently idiomatic javascript way by using array operation some (which is a for loop which will quit looping through if true is returned inside it. Also some will return a false or true value, false is true is not returned). 

Here's an example showing how I would try to make this to be easier to read:

`

const memberEquals = (client: Client, other: Client): boolean =>
  client.member_class === other.member_class &&
  client.member_code === other.member_code &&
  client.instance_id === other.instance_id;

// Filters out clients that have local relatives.
// If the member owning the client or another subsystem
// of the same member is already present locally,
// the client is excluded.
const excludeClientsWithLocalRelatives = (clients: Client[], localClients: Client[]): Client[] => {
  return clients.filter((client: Client) => {
    return !localClients.some( (localClient: Client) => memberEquals(localClient, client))
  });
}
`

If memberEquals (not sure if this is a good function name here) returns true then loop is short circuited and false (negated) is returned to the wrapping filter function. 

I didn't test whether this works by just replacing the existing implementation but it should.

Reply:
Thanks for the suggestion. I updated the implementation accordingly.

I think that `memberEquals` is a descriptive name for the function. However, I added some comments for clarification just in case.